Use ccache in codecov-coverage-report GitHub action

tautschnig opened the pull request:
The GitHub action takes by far the longest, and while the execution time
likely is dominated by test execution we should not unnecessarily waste
time for compiling.

Performance data will be added to the commit message.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

NlightNFotis reviewed (approved):
Absolutely needed, thanks for the addition of this!

tautschnig committed:
Use ccache in codecov-coverage-report GitHub action

The GitHub action takes by far the longest, and while the execution time
likely is dominated by test execution we should not unnecessarily waste
time for compiling.

Building previously took more than 20 minutes, and now takes around 3
minutes instead. The ccache size has to be set to at least 2 GB to avoid
cache entries being removed during build, but the packed archive of
cache results is below 500 MB, which should be safe as far as our use of
the 5 GB of total cache storage is concerned.
